code/modules/recycling/disposal.dm

```diff
@@ -143,6 +143,8 @@
 		return
 
 	if(!I)	return
+	if(I.flags & NODROP)
+		return
 
 	user.drop_item()

```

Comment:
I'm pretty sure that you can get rid of the above check and just have
```
if(!user_drop_item()
     return
```

@TheDZD knows a bit more about the proper unequip usage, especially in relation to NODROP though.